File: back/src/main/java/com/back/global/security/SecurityUser.java
package com.back.global.security;

import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;
import java.util.UUID;

@Getter
public class SecurityUser extends User {
    private final Long id;
    private final UUID uuid;

    public SecurityUser(
            Long id,
            UUID uuid,
            String email,
            Collection<? extends GrantedAuthority> authorities
    ) {
        super(email, "", authorities);
        this.id = id;
        this.uuid = uuid;
    }
}

File: back/src/main/kotlin/com/back/global/security/SecurityUser.kt
package com.back.global.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User
import java.util.UUID

class SecurityUser(
    val id: Long,
    val uuid: UUID,
    email: String,
    authorities: Collection<GrantedAuthority>
) : User(email, "", authorities)
